src/components/Layout/RightPaneLayout/useRightPane.tsx

```diff
@@ -0,0 +1,27 @@
+import { ReactNode, useCallback } from "react";
+import { useRightPaneContext } from "./RightPaneContext";
+
+interface OpenRightPaneOpts {
+  content: ReactNode;
+}
+
+export interface UseRightPaneHook {
+  /** Opens a right pane */
+  // openRightPane: (opts: OpenRightPaneOpts) => void;
+  openRightPane: (opts: OpenRightPaneOpts) => void;
+  /** Closes the right pane */
+  closeRightPane: () => void;
+}
+
+export function useRightPane(): UseRightPaneHook {
+  const { openInPane, closePane } = useRightPaneContext();
+  const openRightPane = useCallback((opts: OpenRightPaneOpts) => {
+    openInPane(opts);
+  }, []);
+  const closeRightPane = useCallback(() => closePane(), []);
```

Comment:
**question:** Do we need to re-create these methods, or can we directly use what we get from the context? It doesn't appear we do anything additional to them, we are just wrapping them in a `useCallback`. Though, they should be stable already as the context memoizes the values. 
...I guess the benefit here is that the we do not define hook dependencies, so even if the context re-creates the `openInPane` or `closePane` methods, then we will not update them here. 🤔  Could that end up causing any issues? 🤔 ... 🚶‍♂️ heading back to the context to comment some more with this thought in mind...

Reply:
Fwiw I'm +1 for keeping the current "this is our public API `useRightPane`" that is separate from the internal `useRightPaneContext`.

Reply:
sounds good to me I agree Id prefer to keep it also

Reply:
I'm good with keeping them separate. My question was can this just do:
```
export function useRightPane(): UseRightPaneHook {
  const { openInPane, closePane } = useRightPaneContext();

  return {
    openRightPane: openInPane,
    closeRightPane: closePane,
  };
```
?
Theres not really a reason, at least that I see, to wrap these function calls in another function.